newdle/client/src/components/Timeline.js

```diff
@@ -233,19 +234,52 @@ function TimelineContent({busySlots, minHour, maxHour}) {
 
 TimelineContent.propTypes = {
   busySlots: PropTypes.array.isRequired,
-  minHour: PropTypes.number,
-  maxHour: PropTypes.number,
-};
-
-TimelineContent.defaultProps = {
-  minHour: 6,
-  maxHour: 24,
+  minHour: PropTypes.number.isRequired,
+  maxHour: PropTypes.number.isRequired,
 };
 
-export default function Timeline({date, availability, minHour, maxHour, hourStep}) {
+export default function Timeline({date, availability, defaultMinHour, defaultMaxHour, hourStep}) {
+  const [minHour, setMinHour] = useState(defaultMinHour);
+  const [maxHour, setMaxHour] = useState(defaultMaxHour);
+  const candidates = useSelector(getTimeslotsForActiveDate);
+  const duration = useSelector(getDuration);
   const hourSeries = _.range(minHour, maxHour + hourStep, hourStep);
   const hourSpan = maxHour - minHour;
+  const defaultHourSpan = defaultMaxHour - defaultMinHour;
   const busySlots = calculateBusyPositions(availability, minHour, maxHour);
+
+  useEffect(() => {
+    const resetHourSpan = () => {
+      setMinHour(defaultMinHour);
+      setMaxHour(defaultMaxHour);
+    };
+
+    const candidatesMoment = candidates.map(c => moment(c, 'HH:mm'));
```

Comment:
We have this `toMoment` function in util/date.js. I guess it would be more consistent to use it also here